### PR TITLE
Making GitHubAction and -parser tests

### PR DESCRIPTION
**Pull request by SebastianWagner2**
Implemented tests for GitHubActions and GitHubActionsParser equivalent to TravisStatus

### CODE REVIEW

File: packages/AutoTDD-Core.package/ATDDNullDataSource.class/instance/repository.st [hunk 1]
```diff
@@ -1,4 +1,4 @@
 accessing
 repository
 	
-	^ String empty
\ No newline at end of file
+	^ ''
```
Comment (benjamin-frost) [inline packages/AutoTDD-Core.package/ATDDNullDataSource.class/instance/repository.st]: What was wrong with `String empty`?

File: packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/active..st [hunk 2]
```diff
@@ -0,0 +1,2 @@
+accessing
+active: aBoolean
```
Comment (benjamin-frost) [inline packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/active..st:2]: This setter has no functionality? Is this correct?

File: packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/config..st [hunk 3]
```diff
@@ -0,0 +1,2 @@
+accessing
+config: anObject
```
Comment (benjamin-frost) [inline packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/config..st:2]: This setter has no functionality? Is this correct?

File: packages/AutoTDD-Dummy-Category.package/ClassGettingTested.class/instance/bla.st [hunk 4]
```diff
@@ -1,4 +1,3 @@
 action
 bla
-
-	ATDDBrowserTests setTestHasRun
\ No newline at end of file
+ATDDBrowserTests setTestHasRun.
```
Comment (benjamin-frost) [inline packages/AutoTDD-Dummy-Category.package/ClassGettingTested.class/instance/bla.st:3]: The formatting looks weird here. Looked more correct before the change..

File: packages/AutoTDD-Tests.package/ATDDGitHubActionTest.class/instance/testConfiguration.st [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+testConfiguration
+	
+	self assert: gitHubAction paused.
+	self assert: (gitHubAction name isKindOf: String).
+	self assert: (gitHubAction repository isKindOf: ATDDNullDataSource).
+	gitHubAction name: 'Hello World'.
+	gitHubAction repositoryURL: self repoURL.
+	"how to test for valid token?, how to test for invalid token?"
+	"gitHubAction token: '9'."
+	"self assert: gitHubAction token = '9'."
```
Comment (benjamin-frost) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionTest.class/instance/testConfiguration.st]: Does this not work?

File: packages/AutoTDD-Tests.package/ATDDNullDataSourceTests.class/instance/testSingleton.st [hunk 6]
```diff
@@ -0,0 +1,7 @@
+tests
+testSingleton
+	
+	| dataSource2 |
```
Comment (benjamin-frost) [inline packages/AutoTDD-Tests.package/ATDDNullDataSourceTests.class/instance/testSingleton.st]: Maybe there is a better name than `dataSource2`?

File: packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st [hunk 7]
```diff
@@ -0,0 +1,11 @@
+as yet unclassified
```
Comment (benjamin-frost) [inline packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st]: classify

File: packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+testConfiguration
+
+	| scrollableTextSpec |
+	
+	"NEWER VERSION: scrollableTextSpec := ATDDPluggableScrollableTextSpec new."
+	scrollableTextSpec := ATDDPluggableTextSpecPlus new.
```
Comment (benjamin-frost) [inline packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st]: The newer version is now merged. Change this if possible

**Review by benjamin-frost — CHANGES_REQUESTED** [review]
Very well done :rocket: 

Please note that none of the test classes have a class comment. Is this the case for the other (old) test classes as well? If they have a comment, we should probably provide one too

File: packages/AutoTDD-Core.package/ATDDNullDataSource.class/instance/repository.st (hunk 1 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Core.package/ATDDNullDataSource.class/instance/repository.st]: Nothing, but we should only use one of them, we can decide for one and change all occurrences sometime

File: packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/active..st (hunk 2 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/active..st:2]: Yeah, NullObjects shouldn't provide any functionality

Comment (SebastianWagner2) [reply packages/AutoTDD-Core.package/ATDDNullTestRunner.class/instance/active..st:2]: Free coverage hellyeah :D

File: packages/AutoTDD-Tests.package/ATDDGitHubActionTest.class/instance/testConfiguration.st (hunk 5 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Tests.package/ATDDGitHubActionTest.class/instance/testConfiguration.st]: Nope

File: packages/AutoTDD-Tests.package/ATDDNullDataSourceTests.class/instance/testSingleton.st (hunk 6 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Tests.package/ATDDNullDataSourceTests.class/instance/testSingleton.st]: Ahhhhh dataSource3 :D

File: packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st (hunk 8 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Tests.package/ATDDPluggableScrollableTextSpecTest.class/instance/testConfiguration.st]: Heeeellyeah

**Comment by SebastianWagner2** [comment]
> 
> 
> Very well done 🚀
> 
> Please note that none of the test classes have a class comment. Is this the case for the other (old) test classes as well? If they have a comment, we should probably provide one too

Lucky For us, the old ones do NOT have comments :P

File: packages/AutoTDD-Core.package/ATDDRemoteParser.class/README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+The ATDDGRemoteParser parses remote testing entity results from data supplied by a RemoteClient.
```
Comment (benjamin-frost) [inline packages/AutoTDD-Core.package/ATDDRemoteParser.class/README.md]:
```suggestion
The ATDDRemoteParser parses remote testing entity results from data supplied by a RemoteClient.
```

**Review by benjamin-frost — CHANGES_REQUESTED** [review]

**Review by benjamin-frost — APPROVED** [review]

**Review by kolioOtSofia — APPROVED** [review]

**Pull request merged**